storage/src/stream.cc

```diff
@@ -555,26 +556,29 @@ unsigned int stream::_find_index_id(
         << service_id << ") (host: " << host_name << ", service: "
         << service_desc << ", special: " << special << ")";
       // Update index_data table.
-      std::ostringstream query;
-      query << "UPDATE " << (db_v2 ? "index_data" : "rt_index_data")
-            << "  SET host_name=:host_name,"
-               "     service_description=:service_description,"
-               "     special=:special"
-               "  WHERE host_id=:host_id"
-               "    AND service_id=:service_id";
+      if (!_index_data_update.prepared()) {
+        std::ostringstream query;
+        query << "UPDATE index_data"
+                 " SET host_name=:host_name,"
+                 " service_description=:service_description,"
+                 " special=:special"
+                 " WHERE host_id=:host_id"
+                 " AND service_id=:service_id";
+        _index_data_update.prepare(query.str());
```

Comment:
Here we do not need anymore to use the ostringstream.
We can just fille the prepare() method with the string.